src/components/Tooltip/EducationalTooltip/BaseEducationalTooltip.tsx

```diff
@@ -1,27 +1,84 @@
 import {NavigationContext} from '@react-navigation/native';
-import React, {memo, useContext, useEffect, useRef, useState} from 'react';
-import type {LayoutRectangle, NativeSyntheticEvent} from 'react-native';
+import React, {memo, useCallback, useContext, useEffect, useLayoutEffect, useRef, useState} from 'react';
+import type {LayoutRectangle, NativeMethods, NativeSyntheticEvent} from 'react-native';
+import {DeviceEventEmitter, Dimensions} from 'react-native';
 import GenericTooltip from '@components/Tooltip/GenericTooltip';
-import type {EducationalTooltipProps} from '@components/Tooltip/types';
-import measureTooltipCoordinate from './measureTooltipCoordinate';
+import type {EducationalTooltipProps, GenericTooltipState} from '@components/Tooltip/types';
+import useSafeAreaInsets from '@hooks/useSafeAreaInsets';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import measureTooltipCoordinate, {getTooltipCoordiate} from './measureTooltipCoordinate';
 
 type LayoutChangeEventWithTarget = NativeSyntheticEvent<{layout: LayoutRectangle; target: HTMLElement}>;
 
+type ScrollingEventData = {
+    isScrolling: boolean;
+};
 /**
  * A component used to wrap an element intended for displaying a tooltip.
  * This tooltip would show immediately without user's interaction and hide after 5 seconds.
  */
-function BaseEducationalTooltip({children, shouldRender = false, shouldHideOnNavigate = true, ...props}: EducationalTooltipProps) {
-    const hideTooltipRef = useRef<() => void>();
+function BaseEducationalTooltip({children, shouldRender = false, shouldHideOnNavigate = true, shouldHideOnScroll = false, ...props}: EducationalTooltipProps) {
+    const genericTooltipStateRef = useRef<GenericTooltipState>();
+    const tooltipElRef = useRef<React.Component & Readonly<NativeMethods>>();
 
     const [shouldMeasure, setShouldMeasure] = useState(false);
     const show = useRef<() => void>();
 
     const navigator = useContext(NavigationContext);
+    const insets = useSafeAreaInsets();
+
+    const setTooltipPosition = useCallback(
+        (isScrolling: boolean) => {
+            if (!shouldHideOnScroll || !genericTooltipStateRef.current || !tooltipElRef.current) {
+                return;
+            }
+
+            const {hideTooltip, showTooltip, updateTargetBounds} = genericTooltipStateRef.current;
+            if (isScrolling) {
+                hideTooltip();
+            } else {
+                getTooltipCoordiate(tooltipElRef.current, (bounds) => {
+                    updateTargetBounds(bounds);
+                    const {y, height} = bounds;
+
+                    const offset = 10; // Buffer space
```

Comment:
What's the buffer space? Can we elaborate it please?

Reply:
Thanks, can we add more context?
```suggestion
                    const offset = 10; // Tooltip hides when content moves 10px past header/footer.
```